### PR TITLE
lthompson/APPEALS-25112

### PR DESCRIPTION
**Pull request by lauren-e-thompson**
Resolves https://jira.devops.va.gov/browse/APPEALS-25112

# Description
## Value Statement:
As a caseflow developer I need to add radio buttons to the team management administrators page, so that I can assign hearings coordinators to be a pexip or a webex conference admin.

## Acceptance Criteria
- [x] Code compiles correctly
- [x] Add Pexip & Webex radio buttons for each user displayed
- [x] Default selection of Pexip

## Testing Plan
- [x] Radio buttons render correctly

## Frontend Display
| Previous master branch UI               | New changes to UI with radio buttons               |
| ---------------------- | ---------------------- |
| Screenshot 2023-07-28 at 2 10 07 PM | Screenshot 2023-07-28 at 3 28 37 PM |

### Test Coverage
Did you include any test coverage for your code? Check below:
- [ ] RSpec
- [x] Jest
- [ ] Other

### Code Climate
Your code does not add any new code climate offenses? If so why?
- [x] No new code climate issues added

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -147,6 +147,11 @@ def can_intake_appeals?
     BvaIntake.singleton.users.include?(self)
   end
 
+  def can_schedule_webex_virtual?
+    @vc_user = User.where(css_id: normalize_css_id)
```
Comment (ThorntonMatthew) [inline app/models/user.rb:151]: `Model.where` returns an array, even if a single record matches the condition.

Use of `User.find_by_css_id` would ensure that either you would get the user you were after, or `nil`.

Comment (ThorntonMatthew) [reply app/models/user.rb:151]: Is there a reason for the instance variable definition here?

I'm wondering if something like the following would fit the bill:
```ruby
def can_schedule_webex_virtual?
  User.find_by_css_id(normalize_css_id)&.meeting_type == Constants.MEETING_TYPES.webex
end
```

Comment (ThorntonMatthew) [reply app/models/user.rb:151]: I just made up the `Constants` bit. I don't actually think that exists.

File: client/app/queue/OrganizationUsers.jsx [hunk 2]
```diff
@@ -38,11 +39,18 @@ const buttonStyle = css({
...
+  display: 'flex',
+  justifyContent: 'space-between',
+  flexWrap: 'wrap'
 });
 const listStyle = css({
   listStyle: 'none'
 });
+const radioContainerStyle = css({
+  padding: '-5rem 5rem 2rem 2rem',
+  // marginTop: '-3.5rem'
```
Comment (ThorntonMatthew) [inline client/app/queue/OrganizationUsers.jsx:52]:
```suggestion
```

File: client/app/queue/SelectConferenceTypeRadioField.jsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import React, { useState } from 'react';
+import { css } from 'glamor';
```
Comment (ThorntonMatthew) [inline client/app/queue/SelectConferenceTypeRadioField.jsx:2]:
```suggestion
```

I don't think this import is in use.

File: client/app/queue/SelectConferenceTypeRadioField.jsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    value: '1'},
+  { displayText: 'Webex',
+    value: '2'}
+];
+
+const SelectConferenceTypeRadioField = ({name}) => {
+  const [value, setValue] = useState("1")
+
+  return (
+    <div>
```
Comment (ThorntonMatthew) [inline client/app/queue/SelectConferenceTypeRadioField.jsx:18]:
```suggestion
    <>
```

You could use a React.Fragment here if you'd like.

File: client/app/queue/SelectConferenceTypeRadioField.jsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+      <RadioField
+        label={COPY.USER_MANAGEMENT_SELECT_HEARINGS_CONFERENCE_TYPE}
+        name={name}
+        options={radioOptions}
+        value={value}
+        onChange={(newValue) => setValue(newValue)}
+        vertical
+    /></div>
+  );
+}
```
Comment (ThorntonMatthew) [inline client/app/queue/SelectConferenceTypeRadioField.jsx:28]:
```suggestion
      />
    </>
  );
}
```
Continuation of https://github.com/department-of-veterans-affairs/caseflow/pull/19057/files#r1280062036

File: client/app/queue/OrganizationUsers.jsx [hunk 6]
```diff
@@ -248,18 +256,29 @@ export default class OrganizationUsers extends React.PureComponent {
...
+              { (judgeTeam || dvcTeam) && admin ?
+                <div {...topUserBorder}></div > :
+                <div {...buttonContainerStyle}>
+                  <div>
+                    { (judgeTeam || dvcTeam) ? '' : this.adminButton(user, admin) }
+                    { this.removeUserButton(user) }
+                  </div>
+                { this.state.organizationName === "Hearing Admin" &&
+                  <div {...radioContainerStyle}>
+                    <SelectConferenceTypeRadioField key={`${user.id}-conference-selection`} name={user.id}/>
```
Comment (ThorntonMatthew) [inline client/app/queue/OrganizationUsers.jsx:276]: Is there a feature toggle that will be used to hide the Pexip/Webex "switches" (like these radio buttons) whenever we eventually turn off Pexip? I'm wondering if these should be conditionally rendered further with that in mind.

File: app/views/queue/index.html.erb [hunk 7]
```diff
@@ -24,6 +24,7 @@
     canEditCavcDashboards: current_user.can_edit_cavc_dashboards?,
     canViewCavcDashboards: current_user.can_view_cavc_dashboards?,
     userIsCobAdmin: ClerkOfTheBoard.singleton.admins.include?(current_user),
+    userCanScheduleWebexVirtual: current_user.can_schedule_webex_virtual?,
```
Comment (ThorntonMatthew) [inline app/views/queue/index.html.erb:27]: This Feature Toggle value likely needs to go into the [UI store](https://github.com/department-of-veterans-affairs/caseflow/blob/66cbeefc444f8dbc995917404ed5f42c494298b2/client/app/queue/reducers.js#L801)'s [reducer](https://github.com/department-of-veterans-affairs/caseflow/blob/473c9101184898cfff10dcfa60362d60d97dfc4a/client/app/queue/uiReducer/uiReducer.js) and have a setter [action](https://github.com/department-of-veterans-affairs/caseflow/blob/473c9101184898cfff10dcfa60362d60d97dfc4a/client/app/queue/uiReducer/uiActions.js) created for it (based on how Queue "primes" its stores).

Comment (ThorntonMatthew) [reply app/views/queue/index.html.erb:27]: This should probably be re-worked, but I don't think it's worth rocking the boat now: https://github.com/department-of-veterans-affairs/caseflow/blob/66cbeefc444f8dbc995917404ed5f42c494298b2/client/app/queue/QueueApp.jsx#L111-L125

File: client/app/queue/OrganizationUsers.jsx (hunk 6 above)
Comment (lauren-e-thompson) [reply client/app/queue/OrganizationUsers.jsx:275]: Resolved - for future work.

**Review by ThorntonMatthew — APPROVED** [review]
Spectacular work!

**Pull request merged**